include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -2837,7 +2837,7 @@ DoFCellAccessor<dimension_, space_dimension_, level_dof_access>::
     *this, dof_indices, this->active_fe_index());
 
   // distribute cell vector
-  global_destination.add(n_dofs, dof_indices.data(), local_source_begin);
+  global_destination.add(n_dofs, dof_indices.data(), &(*local_source_begin));
```

Comment:
This change is wrong. It assumes that the elements pointed to are contiguous, but you are not checking this anywhere. You will be disappointed if you call this function with iterators into `std::list`, for example.